lib/synapse/haproxy.rb

```diff
@@ -743,12 +752,10 @@ def update_backends(watchers)
           rescue StandardError => e
             log.warn "synapse: unknown error writing to socket"
             @restart_required = true
-            return
           else
             unless output == "\n"
```

Comment:
Nit: You could move the "synapse: reconfigured haproxy" log line up to here?

    if output == "\n"
      log.info "synapse: reconfigured haproxy"
    else
      log.warn ...
      @restart_required = true
    end